kouta-backend/src/main/scala/fi/oph/kouta/util/NameHelper.scala

```diff
@@ -1,12 +1,107 @@
 package fi.oph.kouta.util
 
-import fi.oph.kouta.domain.{Fi, Kielistetty, ToteutusMetadata}
+import fi.oph.kouta.domain.{
+  En,
+  Fi,
+  Kieli,
+  Kielistetty,
+  KoulutusMetadata,
+  LukioKoulutusMetadata,
+  LukioToteutusMetadata,
+  LukiolinjaTieto,
+  Sv,
+  Toteutus,
+  ToteutusMetadata
+}
 
 object NameHelper {
+  val EmptyKielistetty: Kielistetty = Map(
+    Fi -> "",
+    Sv -> "",
+    En -> ""
+  )
+  def localizedKielistetty(kielistetty: Option[Kielistetty], kieli: Kieli): String = {
+    kielistetty.getOrElse(EmptyKielistetty).getOrElse(kieli, "")
+  }
+  def isEmptyKielistetty(kielistetty: Option[Kielistetty]): Boolean =
+    kielistetty.isDefined && kielistetty.get.forall { case (lng, translation) => translation.isEmpty }
+  def generateLaajuus(laajuus: Option[String], opintopistetta: Option[Kielistetty], lng: Kieli): String = {
+    if (laajuus.nonEmpty && !isEmptyKielistetty(opintopistetta)) {
+      laajuus.get + " " + localizedKielistetty(opintopistetta, lng)
+    } else {
+      ""
+    }
+  }
+  def getLukiolinjat(toteutus: Toteutus): Option[Seq[LukiolinjaTieto]] = {
+    toteutus.metadata match {
+      case Some(toteutusMetadata) =>
+        toteutusMetadata match {
+          case lukio: LukioToteutusMetadata => Some(lukio.painotukset ++ lukio.erityisetKoulutustehtavat)
+          case _                            => None
+        }
+      case _ => None
+    }
+  }
+  def getLaajuusKoodiUri(koulutusMetadata: Option[KoulutusMetadata]): Option[String] = {
+    koulutusMetadata match {
+      case Some(metadata) =>
+        metadata match {
+          case lukio: LukioKoulutusMetadata => {
+            lukio.opintojenLaajuusKoodiUri match {
+              case Some(laajuus) => Some(laajuus.split("#").head.split('_').last)
+              case _             => None
+            }
+          }
+          case _ => None
+        }
+      case _ => None
+    }
+  }
+  def generateLukioToteutusDisplayName(
+      toteutusMetadata: Option[ToteutusMetadata],
+      koulutusMetadata: Option[KoulutusMetadata],
+      kaannokset: Map[String, Kielistetty],
+      koodiKaannokset: Map[String, Kielistetty]
+  ): Kielistetty = {
+    val yleislinjaNimiOsa = kaannokset.get("toteutuslomake.lukionYleislinjaNimiOsa")
+    val opintopistetta    = kaannokset.get("yleiset.opintopistetta")
+    val lukiolinjat = (for {
+      m <- toteutusMetadata.asInstanceOf[Option[LukioToteutusMetadata]]
+    } yield m.painotukset ++ m.erityisetKoulutustehtavat).getOrElse(Seq())
```

Comment:
Jos tyypityksistä on huolehdittu valmiiksi niin näitä saa yksinkertaistettua, tähän tyyliin: 
```suggestion
      val lukiolinjat = toteutusMetadata.map(t => t.painotukset ++ t.erityisetKoulutustehtavat).getOrElse(Seq())
```